pkg/apis/apm/v1beta1/doc.go

```diff
@@ -0,0 +1,8 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```

Comment:
Why do we need this file rather than just keeping the markers groupversion_info.go?

Reply:
The Kubernetes `gengo` project used by the generator only searches inside the `docs.go` file for the `+groupName`  annotation. If it is not found, the whole package is ignored from the list of "interesting" packages.

Reply:
Ah that's unfortunate, it was kind of nice containing the boilerplate in one place. The groupversion_info.go files are made by the kubebuilder scaffolding so guess they will need to change that too. I also wonder if it is worth opening an issue in controller-tools to add this add it would be useful for kubebuilder people generally. I didn't see any issues open in kubebuilder or controller-tools about adding something similar

Reply:
I suspect that Kubebuilder is not following the conventions of the Kubernetes project. All the other major projects that use `gen-crd-api-references` such as Agones, Kubeflow and KNative don't seem to use Kubebuilder and appear to craft their API by hand. All of their API definitions (and indeed, most Kubernetes APIs) use `doc.go` as the kind of the "entrypoint" to describe the API. 

The gengo project is kind of lagging behind at the moment so my feeling is that it is a bit too early at this point to raise an issue with either project as it is unclear how things are going to evolve.